src/prognostic_equations/forcing/external_forcing.jl

```diff
@@ -243,6 +249,130 @@ function external_forcing_tendency!(Yₜ, Y, p, t, ::GCMForcing)
     return nothing
 end
 
+
+function create_external_forcing_cache(Y)
+    FT = Spaces.undertype(axes(Y.c))
+    ᶜdTdt_fluc = similar(Y.c, FT)
+    ᶜdqtdt_fluc = similar(Y.c, FT)
+    ᶜdTdt_hadv = similar(Y.c, FT)
+    ᶜdqtdt_hadv = similar(Y.c, FT)
+    ᶜdTdt_rad = similar(Y.c, FT)
+    ᶜT_nudge = similar(Y.c, FT)
+    ᶜqt_nudge = similar(Y.c, FT)
+    ᶜu_nudge = similar(Y.c, FT)
+    ᶜv_nudge = similar(Y.c, FT)
+    ᶜinv_τ_wind = similar(Y.c, FT)
+    ᶜinv_τ_scalar = similar(Y.c, FT)
+    ᶜls_subsidence = similar(Y.c, FT)
+    insolation = similar(Fields.level(Y.c.ρ, 1), FT)
+    cos_zenith = similar(Fields.level(Y.c.ρ, 1), FT)
+
+    return (;
+        ᶜdTdt_fluc,
+        ᶜdqtdt_fluc,
+        ᶜdTdt_hadv,
+        ᶜdqtdt_hadv,
+        ᶜdTdt_rad,
+        ᶜT_nudge,
+        ᶜqt_nudge,
+        ᶜu_nudge,
+        ᶜv_nudge,
+        ᶜinv_τ_wind,
+        ᶜinv_τ_scalar,
+        ᶜls_subsidence,
+        insolation,
+        cos_zenith,
+    )
+end
+
+
+function external_forcing_cache(
+    Y,
+    external_forcing::ExternalDrivenTVForcing,
+    params,
+)
+    # current support is for time varying era5 data which does not require vertical advective tendencies
+    # or surface latent and sensible heat fluxes, i.e., the surface state is set with surface temperature 
+    # only. This could be modified to include these terms if needed.
+    (; external_forcing_file, start_date) = external_forcing
+
+    # generate forcing files 
+
+    start_time = DateTime(start_date, "yyyymmdd")
+    column_tendencies = [
+        "ta",
+        "hus",
+        "tntva",
+        "wa",
+        "tntha",
+        "tnhusha",
+        "ua",
+        "va",
+        "tnhusva",
+        "rho",
+        "wap",
+    ]
+    surface_tendencies = ["coszen", "rsdt", "hfls", "hfss", "ts"]
+    column_target_space = axes(similar(Y.c))
+    surface_target_space = axes(similar(Fields.level(Y.c, 1)))
```

Comment:
via ClimaCore level / layer descriptions, `surface` would be the `face` corresponding to the `ClimaCore.Utilities.half` index : e.g. 
`surface_target_space = axes(Fields.level(Y.f.u\_3, Utilities.half))`